include/core/memory.hpp

```diff
@@ -337,40 +340,44 @@ class Memory {
    * \returns a serialize representation of the Memory
    */
   RawMapPair _serializeRaw(char separator = _standardSeparator,
-                           std::size_t lineLength = 64) const;
+                           size_t lineLength = 64) const;
 
   /**
    * \brief returns true if the protection area [protectionBegin, protectionEnd]
    *        overlaps with the area [address, address + amount]
    */
-  bool _overlaps(std::size_t protectionBegin,
-                 std::size_t protectionEnd,
-                 std::size_t address,
-                 std::size_t amount,
+  bool _overlaps(size_t protectionBegin,
+                 size_t protectionEnd,
+                 size_t address,
+                 size_t amount,
                  bool equal) const;
   /**
    * \brief Size of a Byte in Bit
    */
-  std::size_t _byteSize;
+  size_t _byteSize;
   /**
    * \brief Number of Bytes
    */
-  std::size_t _byteCount;
+  size_t _byteCount;
   /**
    * \brief MemoryValue holding all the data
    */
   MemoryValue _data;
   /**
    * \brief This function gets called for every changed area in Memory
    */
-  CallbackFunction _callback;
+  UpdateCallback _callback;
+
+  /** This callback is called when the memory size changes. */
+  SizeCallback _sizeCallback;
 
   /**
    * \brief vector storing data about protected memory areas
-   * \note this takes linear time, one could improve this by implementing some
+   * \note this takes linear time, one could improve this by implementing
+   * some
    *       binary search, or using some indexed algorithm
    */
```

Comment:
Seems like it wasn't formatted properly before and got reformatted automatically on save.